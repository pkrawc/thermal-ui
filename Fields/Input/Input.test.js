import React from 'react'
import renderer from 'react-test-renderer'
import TextInput from '../Input'

describe('Input', () => {
  const props = {
    label: 'input label',
    type: 'text',
    test: {
      id: '1234'
    }
  }
  const tree = renderer.create(<TextInput {...props} />).toJSON()
  // need to set the id's for input and label since they're generated on the fly by shorid
  tree.children[0].props.id = '123'
  tree.children[1].props.htmlFor = '123'
  it('renders without crashing', () => {
    expect(tree).toMatchSnapshot()
  })
})
