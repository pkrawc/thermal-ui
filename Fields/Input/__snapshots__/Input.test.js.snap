exports[`Input renders without crashing 1`] = `
<div
  className="fwjBwY"
  placeholder={undefined}
  value="">
  <input
    className=""
    defaultValue=""
    id="123"
    onChange={[Function]}
    placeholder={undefined}
    type="text" />
  <label
    htmlFor="123">
    input label
  </label>
</div>
`;
